src/doc.rs

```diff
@@ -0,0 +1,354 @@
+use super::ast::{Arg, ArgNames, Statement, TypeAst, TypedModule};
+use itertools::Itertools;
+use std::collections::HashMap;
+#[derive(Debug, PartialEq)]
+struct RawComment {
+  pub line_no: usize,
```

Comment:
We annotate the AST using the byte index rather than a line number so it might be easier to merge the comments in to the AST if store the byte index here rather than the line numbers

Reply:
Yeah, this needs to be updated, it is the byte index* now.

 * At least the char index, which will be correct as long as the file uses the ASCII subset of utf-8.

Reply:
We support unicode so we'll need to check the byte size of the chars as we iterate over them

Reply:
Looks like the offset from the iterator is byte index